src/common/rill-developer-service/MetricsExploreActions.ts

```diff
@@ -34,34 +39,24 @@ export class MetricsExploreActions extends RillDeveloperActions {
     )
       return;
 
-    const model = this.dataModelerStateService
-      .getEntityStateService(EntityType.Model, StateType.Persistent)
-      .getById(rillRequestContext.record.sourceModelId);
-    await Promise.all(
-      expressionEntries.map(async ([id, expression]) => {
-        if (!expression) return;
-        const timeSeries: TimeSeriesRollup =
-          await this.databaseActionQueue.enqueue(
-            {
-              id: metricsDefId,
-              priority: DatabaseActionQueuePriority.ActiveModel,
-            },
-            "generateTimeSeries",
-            [
-              {
-                tableName: model.tableName,
-                timestampColumn: rillRequestContext.record.timeDimension,
-                expression,
-                filters,
-                pixels,
-                rollupInterval,
-              },
-            ]
-          );
-        timeSeries.rollup.id = id;
-        rillRequestContext.actionsChannel.pushMessage(timeSeries.rollup as any);
-      })
-    );
+    if (isolated) {
```

Comment:
hmm, on thinking about this more, hitting this codepath means I have to wait until all the line charts are generated before returning the data. Is that right?

Reply:
This is mainly there for the sparklines in metrics definition editor. We need to isolate them because they can independently update.

The method `generateTimeSeriesForMeasures` pushes each result through the RillActionChannel which in turn will stream it back through the open HTTP connection. So this actually work similar to websocket method.

Reply:
ah got it, thanks for the clarification. I think with this in mind, I still don't understand why `isolated` is needed since if there is only one measure in the arguments it should functionally do the same thing

Reply:
I added in the hopes that sometimes we might update all spark lines (and big numbers) in metrics definition. Especially when the time dimension column is changed. In that case a single call would be made with all measures with isolated = true.